includes/core/classes/class-rest-api.php

```diff
@@ -448,28 +448,43 @@ public function send_emails( int $post_id, array $send, string $message ): bool
 			return false;
 		}
 
-		$members = $this->get_members( $send, $post_id );
-		/* translators: %s: event title. */
-		$subject = sprintf( __( '📅 %s', 'gatherpress' ), get_the_title( $post_id ) );
-		$body    = Utility::render_template(
-			sprintf( '%s/includes/templates/admin/emails/event-email.php', GATHERPRESS_CORE_PATH ),
-			array(
-				'event_id' => $post_id,
-				'message'  => $message,
-			),
-		);
-		$headers = array( 'Content-Type: text/html; charset=UTF-8' );
-		$subject = stripslashes_deep( html_entity_decode( $subject, ENT_QUOTES, 'UTF-8' ) );
+		// Keep the currently logged-in user.
+		$current_user = wp_get_current_user();
 
+		$members = $this->get_members( $send, $post_id );
 		foreach ( $members as $member ) {
 			if ( '0' === get_user_meta( $member->ID, 'gatherpress_event_updates_opt_in', true ) ) {
 				continue;
 			}
-
 			if ( $member->user_email ) {
-				$to = $member->user_email;
+				$to              = $member->user_email;
+				$switched_locale = switch_to_user_locale( $member->ID );
+
+				// Set the current user to the actual member to mail to,
+				// to make sure the GatherPress filters for date- and time- format, as well as the users timezone,
+				// are recognized by the functions inside render_template().
+				wp_set_current_user( $member->ID );
+
+				/* translators: %s: event title. */
+				$subject = sprintf( __( '📅 %s', 'gatherpress' ), get_the_title( $post_id ) );
```

Comment:
This would benefit from additional context for translators (e.g. `_x( '📅 %s', 'email subject', 'gatherpress' )`).

Also worth noting that this is not really useful for translators. What would you even translate here? It's just a placeholder and an emoji.

Reply:
I made an issue out of your idea: https://github.com/GatherPress/gatherpress/issues/827

and will add the context.